src/main/java/frc/robot/util/ControllerHelpers.java

```diff
@@ -9,41 +9,14 @@ public static double deadbandJoystickValue(double joystickValue, double deadband
   }
 
   public static Translation2d fromCircularDiscCoordinates(double x, double y) {
-    // https://stackoverflow.com/a/32391780
-    var mappedX =
-        0.5 * MathHelpers.signedSqrt(2 + Math.pow(x, 2) - Math.pow(y, 2) + 2 * x * Math.sqrt(2))
-            - 0.5
-                * MathHelpers.signedSqrt(
-                    2 + Math.pow(x, 2) - Math.pow(y, 2) - 2 * x * Math.sqrt(2));
-    var mappedY =
-        0.5 * MathHelpers.signedSqrt(2 - Math.pow(x, 2) + Math.pow(y, 2) + 2 * y * Math.sqrt(2))
-            - 0.5
-                * MathHelpers.signedSqrt(
-                    2 - Math.pow(x, 2) + Math.pow(y, 2) - 2 * y * Math.sqrt(2));
 
-    return desaturate(mappedX, mappedY);
-  }
-
-  private static Translation2d desaturate(double x, double y) {
-    double absX = Math.abs(x);
-    double absY = Math.abs(y);
-
-    if (absX > 1 && absX > absY) {
-      // X is too big and is the more problematic one
-
-      var ratio = 1 / absX;
-
-      return new Translation2d(x * ratio, y * ratio);
-    } else if (absY > 1 && absY > absX) {
-      // Y is too big and is the more problematic one
-      var ratio = 1 / absY;
-
-      return new Translation2d(x * ratio, y * ratio);
-
-    } else {
-      //  Everything fine
-      return new Translation2d(x, y);
+    if (Math.abs(x) >= 0.9) {
+      x = Math.copySign(1, x);
+    }
+    if (Math.abs(y) >= 0.9) {
+      y = Math.copySign(1, y);
```

Comment:
this is checking the X & Y components individually, not treating them as two values in a vector & looking at its magnitude